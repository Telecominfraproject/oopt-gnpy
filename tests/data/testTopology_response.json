{
  "response": [
    {
      "response-id": "0",
      "path-properties": {
        "path-metric": [
          {
            "metric-type": "SNR-bandwidth",
            "accumulative-value": 26.75
          },
          {
            "metric-type": "SNR-0.1nm",
            "accumulative-value": 30.84
          },
          {
            "metric-type": "OSNR-bandwidth",
            "accumulative-value": 26.76
          },
          {
            "metric-type": "OSNR-0.1nm",
            "accumulative-value": 30.84
          },
          {
            "metric-type": "reference_power",
            "accumulative-value": 0.001
          },
          {
            "metric-type": "path_bandwidth",
            "accumulative-value": 100000000000.0
          }
        ],
        "path-route-objects": [
          {
            "path-route-object": {
              "index": 0,
              "num-unnum-hop": {
                "node-id": "trx Lorient_KMA",
                "link-tp-id": "trx Lorient_KMA"
              }
            }
          },
          {
            "path-route-object": {
              "index": 1,
              "label-hop": {
                "N": -284,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 2,
              "transponder": {
                "transponder-type": "Voyager",
                "transponder-mode": "mode 1"
              }
            }
          },
          {
            "path-route-object": {
              "index": 3,
              "num-unnum-hop": {
                "node-id": "roadm Lorient_KMA",
                "link-tp-id": "roadm Lorient_KMA"
              }
            }
          },
          {
            "path-route-object": {
              "index": 4,
              "label-hop": {
                "N": -284,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 5,
              "num-unnum-hop": {
                "node-id": "east edfa in Lorient_KMA to Vannes_KBE",
                "link-tp-id": "east edfa in Lorient_KMA to Vannes_KBE"
              }
            }
          },
          {
            "path-route-object": {
              "index": 6,
              "label-hop": {
                "N": -284,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 7,
              "num-unnum-hop": {
                "node-id": "fiber (Lorient_KMA → Vannes_KBE)-F055",
                "link-tp-id": "fiber (Lorient_KMA → Vannes_KBE)-F055"
              }
            }
          },
          {
            "path-route-object": {
              "index": 8,
              "label-hop": {
                "N": -284,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 9,
              "num-unnum-hop": {
                "node-id": "west edfa in Vannes_KBE to Lorient_KMA",
                "link-tp-id": "west edfa in Vannes_KBE to Lorient_KMA"
              }
            }
          },
          {
            "path-route-object": {
              "index": 10,
              "label-hop": {
                "N": -284,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 11,
              "num-unnum-hop": {
                "node-id": "roadm Vannes_KBE",
                "link-tp-id": "roadm Vannes_KBE"
              }
            }
          },
          {
            "path-route-object": {
              "index": 12,
              "label-hop": {
                "N": -284,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 13,
              "num-unnum-hop": {
                "node-id": "trx Vannes_KBE",
                "link-tp-id": "trx Vannes_KBE"
              }
            }
          },
          {
            "path-route-object": {
              "index": 14,
              "label-hop": {
                "N": -284,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 15,
              "transponder": {
                "transponder-type": "Voyager",
                "transponder-mode": "mode 1"
              }
            }
          }
        ]
      }
    },
    {
      "response-id": "1",
      "path-properties": {
        "path-metric": [
          {
            "metric-type": "SNR-bandwidth",
            "accumulative-value": 18.03
          },
          {
            "metric-type": "SNR-0.1nm",
            "accumulative-value": 22.11
          },
          {
            "metric-type": "OSNR-bandwidth",
            "accumulative-value": 18.57
          },
          {
            "metric-type": "OSNR-0.1nm",
            "accumulative-value": 22.65
          },
          {
            "metric-type": "reference_power",
            "accumulative-value": 0.0012589254117941673
          },
          {
            "metric-type": "path_bandwidth",
            "accumulative-value": 10000000000.0
          }
        ],
        "path-route-objects": [
          {
            "path-route-object": {
              "index": 0,
              "num-unnum-hop": {
                "node-id": "trx Brest_KLA",
                "link-tp-id": "trx Brest_KLA"
              }
            }
          },
          {
            "path-route-object": {
              "index": 1,
              "label-hop": {
                "N": -276,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 2,
              "transponder": {
                "transponder-type": "Voyager",
                "transponder-mode": "mode 1"
              }
            }
          },
          {
            "path-route-object": {
              "index": 3,
              "num-unnum-hop": {
                "node-id": "roadm Brest_KLA",
                "link-tp-id": "roadm Brest_KLA"
              }
            }
          },
          {
            "path-route-object": {
              "index": 4,
              "label-hop": {
                "N": -276,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 5,
              "num-unnum-hop": {
                "node-id": "east edfa in Brest_KLA to Morlaix",
                "link-tp-id": "east edfa in Brest_KLA to Morlaix"
              }
            }
          },
          {
            "path-route-object": {
              "index": 6,
              "label-hop": {
                "N": -276,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 7,
              "num-unnum-hop": {
                "node-id": "fiber (Brest_KLA → Morlaix)-F060",
                "link-tp-id": "fiber (Brest_KLA → Morlaix)-F060"
              }
            }
          },
          {
            "path-route-object": {
              "index": 8,
              "label-hop": {
                "N": -276,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 9,
              "num-unnum-hop": {
                "node-id": "east fused spans in Morlaix",
                "link-tp-id": "east fused spans in Morlaix"
              }
            }
          },
          {
            "path-route-object": {
              "index": 10,
              "label-hop": {
                "N": -276,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 11,
              "num-unnum-hop": {
                "node-id": "fiber (Morlaix → Lannion_CAS)-F059",
                "link-tp-id": "fiber (Morlaix → Lannion_CAS)-F059"
              }
            }
          },
          {
            "path-route-object": {
              "index": 12,
              "label-hop": {
                "N": -276,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 13,
              "num-unnum-hop": {
                "node-id": "west edfa in Lannion_CAS to Morlaix",
                "link-tp-id": "west edfa in Lannion_CAS to Morlaix"
              }
            }
          },
          {
            "path-route-object": {
              "index": 14,
              "label-hop": {
                "N": -276,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 15,
              "num-unnum-hop": {
                "node-id": "roadm Lannion_CAS",
                "link-tp-id": "roadm Lannion_CAS"
              }
            }
          },
          {
            "path-route-object": {
              "index": 16,
              "label-hop": {
                "N": -276,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 17,
              "num-unnum-hop": {
                "node-id": "east edfa in Lannion_CAS to Corlay",
                "link-tp-id": "east edfa in Lannion_CAS to Corlay"
              }
            }
          },
          {
            "path-route-object": {
              "index": 18,
              "label-hop": {
                "N": -276,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 19,
              "num-unnum-hop": {
                "node-id": "fiber (Lannion_CAS → Corlay)-F061",
                "link-tp-id": "fiber (Lannion_CAS → Corlay)-F061"
              }
            }
          },
          {
            "path-route-object": {
              "index": 20,
              "label-hop": {
                "N": -276,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 21,
              "num-unnum-hop": {
                "node-id": "west fused spans in Corlay",
                "link-tp-id": "west fused spans in Corlay"
              }
            }
          },
          {
            "path-route-object": {
              "index": 22,
              "label-hop": {
                "N": -276,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 23,
              "num-unnum-hop": {
                "node-id": "fiber (Corlay → Loudeac)-F010",
                "link-tp-id": "fiber (Corlay → Loudeac)-F010"
              }
            }
          },
          {
            "path-route-object": {
              "index": 24,
              "label-hop": {
                "N": -276,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 25,
              "num-unnum-hop": {
                "node-id": "west fused spans in Loudeac",
                "link-tp-id": "west fused spans in Loudeac"
              }
            }
          },
          {
            "path-route-object": {
              "index": 26,
              "label-hop": {
                "N": -276,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 27,
              "num-unnum-hop": {
                "node-id": "fiber (Loudeac → Lorient_KMA)-F054",
                "link-tp-id": "fiber (Loudeac → Lorient_KMA)-F054"
              }
            }
          },
          {
            "path-route-object": {
              "index": 28,
              "label-hop": {
                "N": -276,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 29,
              "num-unnum-hop": {
                "node-id": "west edfa in Lorient_KMA to Loudeac",
                "link-tp-id": "west edfa in Lorient_KMA to Loudeac"
              }
            }
          },
          {
            "path-route-object": {
              "index": 30,
              "label-hop": {
                "N": -276,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 31,
              "num-unnum-hop": {
                "node-id": "roadm Lorient_KMA",
                "link-tp-id": "roadm Lorient_KMA"
              }
            }
          },
          {
            "path-route-object": {
              "index": 32,
              "label-hop": {
                "N": -276,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 33,
              "num-unnum-hop": {
                "node-id": "east edfa in Lorient_KMA to Vannes_KBE",
                "link-tp-id": "east edfa in Lorient_KMA to Vannes_KBE"
              }
            }
          },
          {
            "path-route-object": {
              "index": 34,
              "label-hop": {
                "N": -276,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 35,
              "num-unnum-hop": {
                "node-id": "fiber (Lorient_KMA → Vannes_KBE)-F055",
                "link-tp-id": "fiber (Lorient_KMA → Vannes_KBE)-F055"
              }
            }
          },
          {
            "path-route-object": {
              "index": 36,
              "label-hop": {
                "N": -276,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 37,
              "num-unnum-hop": {
                "node-id": "west edfa in Vannes_KBE to Lorient_KMA",
                "link-tp-id": "west edfa in Vannes_KBE to Lorient_KMA"
              }
            }
          },
          {
            "path-route-object": {
              "index": 38,
              "label-hop": {
                "N": -276,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 39,
              "num-unnum-hop": {
                "node-id": "roadm Vannes_KBE",
                "link-tp-id": "roadm Vannes_KBE"
              }
            }
          },
          {
            "path-route-object": {
              "index": 40,
              "label-hop": {
                "N": -276,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 41,
              "num-unnum-hop": {
                "node-id": "trx Vannes_KBE",
                "link-tp-id": "trx Vannes_KBE"
              }
            }
          },
          {
            "path-route-object": {
              "index": 42,
              "label-hop": {
                "N": -276,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 43,
              "transponder": {
                "transponder-type": "Voyager",
                "transponder-mode": "mode 1"
              }
            }
          }
        ]
      }
    },
    {
      "response-id": "3",
      "path-properties": {
        "path-metric": [
          {
            "metric-type": "SNR-bandwidth",
            "accumulative-value": 21.77
          },
          {
            "metric-type": "SNR-0.1nm",
            "accumulative-value": 25.85
          },
          {
            "metric-type": "OSNR-bandwidth",
            "accumulative-value": 24.2
          },
          {
            "metric-type": "OSNR-0.1nm",
            "accumulative-value": 28.29
          },
          {
            "metric-type": "reference_power",
            "accumulative-value": 0.0012589254117941673
          },
          {
            "metric-type": "path_bandwidth",
            "accumulative-value": 60000000000.0
          }
        ],
        "path-route-objects": [
          {
            "path-route-object": {
              "index": 0,
              "num-unnum-hop": {
                "node-id": "trx Lannion_CAS",
                "link-tp-id": "trx Lannion_CAS"
              }
            }
          },
          {
            "path-route-object": {
              "index": 1,
              "label-hop": {
                "N": -284,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 2,
              "transponder": {
                "transponder-type": "vendorA_trx-type1",
                "transponder-mode": "mode 1"
              }
            }
          },
          {
            "path-route-object": {
              "index": 3,
              "num-unnum-hop": {
                "node-id": "roadm Lannion_CAS",
                "link-tp-id": "roadm Lannion_CAS"
              }
            }
          },
          {
            "path-route-object": {
              "index": 4,
              "label-hop": {
                "N": -284,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 5,
              "num-unnum-hop": {
                "node-id": "east edfa in Lannion_CAS to Stbrieuc",
                "link-tp-id": "east edfa in Lannion_CAS to Stbrieuc"
              }
            }
          },
          {
            "path-route-object": {
              "index": 6,
              "label-hop": {
                "N": -284,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 7,
              "num-unnum-hop": {
                "node-id": "fiber (Lannion_CAS → Stbrieuc)-F056",
                "link-tp-id": "fiber (Lannion_CAS → Stbrieuc)-F056"
              }
            }
          },
          {
            "path-route-object": {
              "index": 8,
              "label-hop": {
                "N": -284,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 9,
              "num-unnum-hop": {
                "node-id": "east edfa in Stbrieuc to Rennes_STA",
                "link-tp-id": "east edfa in Stbrieuc to Rennes_STA"
              }
            }
          },
          {
            "path-route-object": {
              "index": 10,
              "label-hop": {
                "N": -284,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 11,
              "num-unnum-hop": {
                "node-id": "fiber (Stbrieuc → Rennes_STA)-F057",
                "link-tp-id": "fiber (Stbrieuc → Rennes_STA)-F057"
              }
            }
          },
          {
            "path-route-object": {
              "index": 12,
              "label-hop": {
                "N": -284,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 13,
              "num-unnum-hop": {
                "node-id": "west edfa in Rennes_STA to Stbrieuc",
                "link-tp-id": "west edfa in Rennes_STA to Stbrieuc"
              }
            }
          },
          {
            "path-route-object": {
              "index": 14,
              "label-hop": {
                "N": -284,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 15,
              "num-unnum-hop": {
                "node-id": "roadm Rennes_STA",
                "link-tp-id": "roadm Rennes_STA"
              }
            }
          },
          {
            "path-route-object": {
              "index": 16,
              "label-hop": {
                "N": -284,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 17,
              "num-unnum-hop": {
                "node-id": "trx Rennes_STA",
                "link-tp-id": "trx Rennes_STA"
              }
            }
          },
          {
            "path-route-object": {
              "index": 18,
              "label-hop": {
                "N": -284,
                "M": 4
              }
            }
          },
          {
            "path-route-object": {
              "index": 19,
              "transponder": {
                "transponder-type": "vendorA_trx-type1",
                "transponder-mode": "mode 1"
              }
            }
          }
        ]
      }
    },
    {
      "response-id": "4",
      "path-properties": {
        "path-metric": [
          {
            "metric-type": "SNR-bandwidth",
            "accumulative-value": 15.05
          },
          {
            "metric-type": "SNR-0.1nm",
            "accumulative-value": 22.15
          },
          {
            "metric-type": "OSNR-bandwidth",
            "accumulative-value": 15.18
          },
          {
            "metric-type": "OSNR-0.1nm",
            "accumulative-value": 22.27
          },
          {
            "metric-type": "reference_power",
            "accumulative-value": 0.001
          },
          {
            "metric-type": "path_bandwidth",
            "accumulative-value": 150000000000.0
          }
        ],
        "path-route-objects": [
          {
            "path-route-object": {
              "index": 0,
              "num-unnum-hop": {
                "node-id": "trx Rennes_STA",
                "link-tp-id": "trx Rennes_STA"
              }
            }
          },
          {
            "path-route-object": {
              "index": 1,
              "label-hop": {
                "N": -266,
                "M": 6
              }
            }
          },
          {
            "path-route-object": {
              "index": 2,
              "transponder": {
                "transponder-type": "vendorA_trx-type1",
                "transponder-mode": "mode 2"
              }
            }
          },
          {
            "path-route-object": {
              "index": 3,
              "num-unnum-hop": {
                "node-id": "roadm Rennes_STA",
                "link-tp-id": "roadm Rennes_STA"
              }
            }
          },
          {
            "path-route-object": {
              "index": 4,
              "label-hop": {
                "N": -266,
                "M": 6
              }
            }
          },
          {
            "path-route-object": {
              "index": 5,
              "num-unnum-hop": {
                "node-id": "east edfa in Rennes_STA to Ploermel",
                "link-tp-id": "east edfa in Rennes_STA to Ploermel"
              }
            }
          },
          {
            "path-route-object": {
              "index": 6,
              "label-hop": {
                "N": -266,
                "M": 6
              }
            }
          },
          {
            "path-route-object": {
              "index": 7,
              "num-unnum-hop": {
                "node-id": "fiber (Rennes_STA → Ploermel)-",
                "link-tp-id": "fiber (Rennes_STA → Ploermel)-"
              }
            }
          },
          {
            "path-route-object": {
              "index": 8,
              "label-hop": {
                "N": -266,
                "M": 6
              }
            }
          },
          {
            "path-route-object": {
              "index": 9,
              "num-unnum-hop": {
                "node-id": "east edfa in Ploermel to Vannes_KBE",
                "link-tp-id": "east edfa in Ploermel to Vannes_KBE"
              }
            }
          },
          {
            "path-route-object": {
              "index": 10,
              "label-hop": {
                "N": -266,
                "M": 6
              }
            }
          },
          {
            "path-route-object": {
              "index": 11,
              "num-unnum-hop": {
                "node-id": "fiber (Ploermel → Vannes_KBE)-",
                "link-tp-id": "fiber (Ploermel → Vannes_KBE)-"
              }
            }
          },
          {
            "path-route-object": {
              "index": 12,
              "label-hop": {
                "N": -266,
                "M": 6
              }
            }
          },
          {
            "path-route-object": {
              "index": 13,
              "num-unnum-hop": {
                "node-id": "west edfa in Vannes_KBE to Ploermel",
                "link-tp-id": "west edfa in Vannes_KBE to Ploermel"
              }
            }
          },
          {
            "path-route-object": {
              "index": 14,
              "label-hop": {
                "N": -266,
                "M": 6
              }
            }
          },
          {
            "path-route-object": {
              "index": 15,
              "num-unnum-hop": {
                "node-id": "roadm Vannes_KBE",
                "link-tp-id": "roadm Vannes_KBE"
              }
            }
          },
          {
            "path-route-object": {
              "index": 16,
              "label-hop": {
                "N": -266,
                "M": 6
              }
            }
          },
          {
            "path-route-object": {
              "index": 17,
              "num-unnum-hop": {
                "node-id": "east edfa in Vannes_KBE to Lorient_KMA",
                "link-tp-id": "east edfa in Vannes_KBE to Lorient_KMA"
              }
            }
          },
          {
            "path-route-object": {
              "index": 18,
              "label-hop": {
                "N": -266,
                "M": 6
              }
            }
          },
          {
            "path-route-object": {
              "index": 19,
              "num-unnum-hop": {
                "node-id": "fiber (Vannes_KBE → Lorient_KMA)-F055",
                "link-tp-id": "fiber (Vannes_KBE → Lorient_KMA)-F055"
              }
            }
          },
          {
            "path-route-object": {
              "index": 20,
              "label-hop": {
                "N": -266,
                "M": 6
              }
            }
          },
          {
            "path-route-object": {
              "index": 21,
              "num-unnum-hop": {
                "node-id": "west edfa in Lorient_KMA to Vannes_KBE",
                "link-tp-id": "west edfa in Lorient_KMA to Vannes_KBE"
              }
            }
          },
          {
            "path-route-object": {
              "index": 22,
              "label-hop": {
                "N": -266,
                "M": 6
              }
            }
          },
          {
            "path-route-object": {
              "index": 23,
              "num-unnum-hop": {
                "node-id": "roadm Lorient_KMA",
                "link-tp-id": "roadm Lorient_KMA"
              }
            }
          },
          {
            "path-route-object": {
              "index": 24,
              "label-hop": {
                "N": -266,
                "M": 6
              }
            }
          },
          {
            "path-route-object": {
              "index": 25,
              "num-unnum-hop": {
                "node-id": "east edfa in Lorient_KMA to Loudeac",
                "link-tp-id": "east edfa in Lorient_KMA to Loudeac"
              }
            }
          },
          {
            "path-route-object": {
              "index": 26,
              "label-hop": {
                "N": -266,
                "M": 6
              }
            }
          },
          {
            "path-route-object": {
              "index": 27,
              "num-unnum-hop": {
                "node-id": "fiber (Lorient_KMA → Loudeac)-F054",
                "link-tp-id": "fiber (Lorient_KMA → Loudeac)-F054"
              }
            }
          },
          {
            "path-route-object": {
              "index": 28,
              "label-hop": {
                "N": -266,
                "M": 6
              }
            }
          },
          {
            "path-route-object": {
              "index": 29,
              "num-unnum-hop": {
                "node-id": "east fused spans in Loudeac",
                "link-tp-id": "east fused spans in Loudeac"
              }
            }
          },
          {
            "path-route-object": {
              "index": 30,
              "label-hop": {
                "N": -266,
                "M": 6
              }
            }
          },
          {
            "path-route-object": {
              "index": 31,
              "num-unnum-hop": {
                "node-id": "fiber (Loudeac → Corlay)-F010",
                "link-tp-id": "fiber (Loudeac → Corlay)-F010"
              }
            }
          },
          {
            "path-route-object": {
              "index": 32,
              "label-hop": {
                "N": -266,
                "M": 6
              }
            }
          },
          {
            "path-route-object": {
              "index": 33,
              "num-unnum-hop": {
                "node-id": "east fused spans in Corlay",
                "link-tp-id": "east fused spans in Corlay"
              }
            }
          },
          {
            "path-route-object": {
              "index": 34,
              "label-hop": {
                "N": -266,
                "M": 6
              }
            }
          },
          {
            "path-route-object": {
              "index": 35,
              "num-unnum-hop": {
                "node-id": "fiber (Corlay → Lannion_CAS)-F061",
                "link-tp-id": "fiber (Corlay → Lannion_CAS)-F061"
              }
            }
          },
          {
            "path-route-object": {
              "index": 36,
              "label-hop": {
                "N": -266,
                "M": 6
              }
            }
          },
          {
            "path-route-object": {
              "index": 37,
              "num-unnum-hop": {
                "node-id": "west edfa in Lannion_CAS to Corlay",
                "link-tp-id": "west edfa in Lannion_CAS to Corlay"
              }
            }
          },
          {
            "path-route-object": {
              "index": 38,
              "label-hop": {
                "N": -266,
                "M": 6
              }
            }
          },
          {
            "path-route-object": {
              "index": 39,
              "num-unnum-hop": {
                "node-id": "roadm Lannion_CAS",
                "link-tp-id": "roadm Lannion_CAS"
              }
            }
          },
          {
            "path-route-object": {
              "index": 40,
              "label-hop": {
                "N": -266,
                "M": 6
              }
            }
          },
          {
            "path-route-object": {
              "index": 41,
              "num-unnum-hop": {
                "node-id": "trx Lannion_CAS",
                "link-tp-id": "trx Lannion_CAS"
              }
            }
          },
          {
            "path-route-object": {
              "index": 42,
              "label-hop": {
                "N": -266,
                "M": 6
              }
            }
          },
          {
            "path-route-object": {
              "index": 43,
              "transponder": {
                "transponder-type": "vendorA_trx-type1",
                "transponder-mode": "mode 2"
              }
            }
          }
        ]
      }
    },
    {
      "response-id": "5",
      "path-properties": {
        "path-metric": [
          {
            "metric-type": "SNR-bandwidth",
            "accumulative-value": 21.68
          },
          {
            "metric-type": "SNR-0.1nm",
            "accumulative-value": 28.78
          },
          {
            "metric-type": "OSNR-bandwidth",
            "accumulative-value": 23.7
          },
          {
            "metric-type": "OSNR-0.1nm",
            "accumulative-value": 30.79
          },
          {
            "metric-type": "reference_power",
            "accumulative-value": 0.0019952623149688794
          },
          {
            "metric-type": "path_bandwidth",
            "accumulative-value": 20000000000.0
          }
        ],
        "path-route-objects": [
          {
            "path-route-object": {
              "index": 0,
              "num-unnum-hop": {
                "node-id": "trx Rennes_STA",
                "link-tp-id": "trx Rennes_STA"
              }
            }
          },
          {
            "path-route-object": {
              "index": 1,
              "label-hop": {
                "N": -274,
                "M": 6
              }
            }
          },
          {
            "path-route-object": {
              "index": 2,
              "transponder": {
                "transponder-type": "vendorA_trx-type1",
                "transponder-mode": "mode 2"
              }
            }
          },
          {
            "path-route-object": {
              "index": 3,
              "num-unnum-hop": {
                "node-id": "roadm Rennes_STA",
                "link-tp-id": "roadm Rennes_STA"
              }
            }
          },
          {
            "path-route-object": {
              "index": 4,
              "label-hop": {
                "N": -274,
                "M": 6
              }
            }
          },
          {
            "path-route-object": {
              "index": 5,
              "num-unnum-hop": {
                "node-id": "east edfa in Rennes_STA to Stbrieuc",
                "link-tp-id": "east edfa in Rennes_STA to Stbrieuc"
              }
            }
          },
          {
            "path-route-object": {
              "index": 6,
              "label-hop": {
                "N": -274,
                "M": 6
              }
            }
          },
          {
            "path-route-object": {
              "index": 7,
              "num-unnum-hop": {
                "node-id": "fiber (Rennes_STA → Stbrieuc)-F057",
                "link-tp-id": "fiber (Rennes_STA → Stbrieuc)-F057"
              }
            }
          },
          {
            "path-route-object": {
              "index": 8,
              "label-hop": {
                "N": -274,
                "M": 6
              }
            }
          },
          {
            "path-route-object": {
              "index": 9,
              "num-unnum-hop": {
                "node-id": "west edfa in Stbrieuc to Rennes_STA",
                "link-tp-id": "west edfa in Stbrieuc to Rennes_STA"
              }
            }
          },
          {
            "path-route-object": {
              "index": 10,
              "label-hop": {
                "N": -274,
                "M": 6
              }
            }
          },
          {
            "path-route-object": {
              "index": 11,
              "num-unnum-hop": {
                "node-id": "fiber (Stbrieuc → Lannion_CAS)-F056",
                "link-tp-id": "fiber (Stbrieuc → Lannion_CAS)-F056"
              }
            }
          },
          {
            "path-route-object": {
              "index": 12,
              "label-hop": {
                "N": -274,
                "M": 6
              }
            }
          },
          {
            "path-route-object": {
              "index": 13,
              "num-unnum-hop": {
                "node-id": "west edfa in Lannion_CAS to Stbrieuc",
                "link-tp-id": "west edfa in Lannion_CAS to Stbrieuc"
              }
            }
          },
          {
            "path-route-object": {
              "index": 14,
              "label-hop": {
                "N": -274,
                "M": 6
              }
            }
          },
          {
            "path-route-object": {
              "index": 15,
              "num-unnum-hop": {
                "node-id": "roadm Lannion_CAS",
                "link-tp-id": "roadm Lannion_CAS"
              }
            }
          },
          {
            "path-route-object": {
              "index": 16,
              "label-hop": {
                "N": -274,
                "M": 6
              }
            }
          },
          {
            "path-route-object": {
              "index": 17,
              "num-unnum-hop": {
                "node-id": "trx Lannion_CAS",
                "link-tp-id": "trx Lannion_CAS"
              }
            }
          },
          {
            "path-route-object": {
              "index": 18,
              "label-hop": {
                "N": -274,
                "M": 6
              }
            }
          },
          {
            "path-route-object": {
              "index": 19,
              "transponder": {
                "transponder-type": "vendorA_trx-type1",
                "transponder-mode": "mode 2"
              }
            }
          }
        ]
      }
    },
    {
      "response-id": "6",
      "no-path": {
        "no-path": "NO_PATH"
      }
    }
  ]
}
