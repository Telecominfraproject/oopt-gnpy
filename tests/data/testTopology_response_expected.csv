response-id,source,destination,path_bandwidth,Pass?,nb of tsp pairs,total cost,transponder-type,transponder-mode,OSNR-0.1nm,SNR-0.1nm,SNR-bandwidth,baud rate (Gbaud),input power (dBm),path,"spectrum (N,M)",reversed path OSNR-0.1nm,reversed path SNR-0.1nm,reversed path SNR-bandwidth
0,trx Lorient_KMA,trx Vannes_KBE,100.0,True,1,1,Voyager,mode 1,30.84,30.84,26.75,32.0,0.0,trx Lorient_KMA | roadm Lorient_KMA | east edfa in Lorient_KMA to Vannes_KBE | fiber (Lorient_KMA → Vannes_KBE)-F055 | west edfa in Vannes_KBE to Lorient_KMA | roadm Vannes_KBE | trx Vannes_KBE,"-284, 4",,,
1,trx Brest_KLA,trx Vannes_KBE,10.0,True,1,1,Voyager,mode 1,22.65,22.11,18.03,32.0,1.0,trx Brest_KLA | roadm Brest_KLA | east edfa in Brest_KLA to Morlaix | fiber (Brest_KLA → Morlaix)-F060 | east fused spans in Morlaix | fiber (Morlaix → Lannion_CAS)-F059 | west edfa in Lannion_CAS to Morlaix | roadm Lannion_CAS | east edfa in Lannion_CAS to Corlay | fiber (Lannion_CAS → Corlay)-F061 | west fused spans in Corlay | fiber (Corlay → Loudeac)-F010 | west fused spans in Loudeac | fiber (Loudeac → Lorient_KMA)-F054 | west edfa in Lorient_KMA to Loudeac | roadm Lorient_KMA | east edfa in Lorient_KMA to Vannes_KBE | fiber (Lorient_KMA → Vannes_KBE)-F055 | west edfa in Vannes_KBE to Lorient_KMA | roadm Vannes_KBE | trx Vannes_KBE,"-276, 4",,,
3,trx Lannion_CAS,trx Rennes_STA,60.0,True,1,1,vendorA_trx-type1,mode 1,28.29,25.85,21.77,32.0,1.0,trx Lannion_CAS | roadm Lannion_CAS | east edfa in Lannion_CAS to Stbrieuc | fiber (Lannion_CAS → Stbrieuc)-F056 | east edfa in Stbrieuc to Rennes_STA | fiber (Stbrieuc → Rennes_STA)-F057 | west edfa in Rennes_STA to Stbrieuc | roadm Rennes_STA | trx Rennes_STA,"-284, 4",,,
4,trx Rennes_STA,trx Lannion_CAS,150.0,True,1,1,vendorA_trx-type1,mode 2,22.27,22.15,15.05,64.0,0.0,trx Rennes_STA | roadm Rennes_STA | east edfa in Rennes_STA to Ploermel | fiber (Rennes_STA → Ploermel)- | east edfa in Ploermel to Vannes_KBE | fiber (Ploermel → Vannes_KBE)- | west edfa in Vannes_KBE to Ploermel | roadm Vannes_KBE | east edfa in Vannes_KBE to Lorient_KMA | fiber (Vannes_KBE → Lorient_KMA)-F055 | west edfa in Lorient_KMA to Vannes_KBE | roadm Lorient_KMA | east edfa in Lorient_KMA to Loudeac | fiber (Lorient_KMA → Loudeac)-F054 | east fused spans in Loudeac | fiber (Loudeac → Corlay)-F010 | east fused spans in Corlay | fiber (Corlay → Lannion_CAS)-F061 | west edfa in Lannion_CAS to Corlay | roadm Lannion_CAS | trx Lannion_CAS,"-266, 6",,,
5,trx Rennes_STA,trx Lannion_CAS,20.0,True,1,1,vendorA_trx-type1,mode 2,30.79,28.78,21.68,64.0,3.0,trx Rennes_STA | roadm Rennes_STA | east edfa in Rennes_STA to Stbrieuc | fiber (Rennes_STA → Stbrieuc)-F057 | west edfa in Stbrieuc to Rennes_STA | fiber (Stbrieuc → Lannion_CAS)-F056 | west edfa in Lannion_CAS to Stbrieuc | roadm Lannion_CAS | trx Lannion_CAS,"-274, 6",,,
6,,,,NO_PATH,,,,,,,,,,,,,,
